src/appleseed/renderer/modeling/object/meshobjectprimitives.cpp

```diff
@@ -0,0 +1,251 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2016 Ramon Blanquer
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "meshobjectprimitives.h"
+
+// appleseed.renderer headers.
+#include "renderer/global/globaltypes.h"
+#include "renderer/modeling/object/meshobject.h"
+#include "renderer/modeling/object/meshobjectoperations.h"
+#include "renderer/modeling/object/triangle.h"
+#include "renderer/utility/paramarray.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/scalar.h"
+#include "foundation/math/vector.h"
+#include "foundation/platform/types.h"
+
+// Standard headers.
+#include <cmath>
+
+using namespace foundation;
+using namespace std;
+
+namespace renderer
+{
+
+class ParametricSphere
+{
+public:
+    explicit ParametricSphere(const double radius)
+      : m_radius(radius)
+    {
+    }
+
+    GVector3 evaluate(const double u, const double v) const
+    {
+        const double theta = 2.0 * Pi * u;
```

Comment:
You can use the `TwoPi` constant to save one multiplication here (this won't necessarily be done by the compiler, depending on the floating point mode).